flathunter/filter.py

```diff
@@ -172,30 +176,65 @@ def is_interesting(self, expose):
         return pps <= self.max_pps
 
 
-class FilterBuilder:
+class DistanceFilter(AbstractFilter):
+    """Exclude properties based on distance or duration to a location
+    
+    This must be in the post-processing filter chain, as it requires data
+    from the Google Maps API, which is not available right after scraping."""
+    
+    distance_config: DistanceConfig
+    
+    def __init__(self, distance_config: DistanceConfig):
+        self.distance_config = distance_config
+    
+    def is_interesting(self, expose):
+        durations: Dict[str, DistanceElement] = expose.get('durations_unformatted', None)
+        if durations is None or self.distance_config.location_name not in durations:
+            logger.info('DurationFilter is enabled, but no GMaps data found. Skipping filter.')
+            return True
+        distance = durations[self.distance_config.location_name].distance.meters
+        duration = durations[self.distance_config.location_name].duration.seconds
+        out = True
+        if self.distance_config.max_distance_meters:
+            out &= distance < self.distance_config.max_distance_meters
+        if self.distance_config.max_duration_seconds:
+            out &= duration < self.distance_config.max_duration_seconds
+        return out
+
+
+class FilterChainBuilder:
     """Construct a filter chain"""
     filters: List[AbstractFilter]
 
     def __init__(self):
         self.filters = []
 
-    def _append_filter_if_not_empty(self, filter_class: ABCMeta, filter_config: Any):
+    def _append_filter_if_not_empty(
+            self,
+            filter_class: ABCMeta, 
+            filter_config: Any):
         """Appends a filter to the list if its configuration is set"""
         if not filter_config:
             return
         self.filters.append(filter_class(filter_config))
 
-    def read_config(self, config):
+    def read_config(self, config, filter_chain: FilterChainName):
         """Adds filters from a config dictionary"""
-        self._append_filter_if_not_empty(TitleFilter, config.excluded_titles())
-        self._append_filter_if_not_empty(MinPriceFilter, config.min_price())
-        self._append_filter_if_not_empty(MaxPriceFilter, config.max_price())
-        self._append_filter_if_not_empty(MinSizeFilter, config.min_size())
-        self._append_filter_if_not_empty(MaxSizeFilter, config.max_size())
-        self._append_filter_if_not_empty(MinRoomsFilter, config.min_rooms())
-        self._append_filter_if_not_empty(MaxRoomsFilter, config.max_rooms())
-        self._append_filter_if_not_empty(
-            PPSFilter, config.max_price_per_square())
+        if filter_chain == FilterChainName.preprocess:
```

Comment:
I'm a bit "meh" on if-else chains. Do you think we could do this with Enum and Match? 

https://realpython.com/python-enum/#using-enumerations-in-if-and-match-statements